File: src/2025/day1/day1.c
#include <stdio.h>
#include <stdlib.h>

int mod(int a, int n) {
  int r = a % n;
  if (r < 0) {
    r += n;
  }
  return r;
}

int main() {
  char* file_name = "2025_day1.txt";
  FILE* fp = fopen(file_name, "r");
  char line[32];
  char direction;
  int distance;
  int position = 50;
  if (fp == NULL) {
    printf("Failed to read the file\n");
    return 1;
  }
  int sum = 0;
  while (fgets(line, sizeof(line), fp)) {
    direction = line[0];
    distance = atoi(&line[1]);
    if (direction == 'L') {
      position = mod(position - distance, 100);
    } else if (direction == 'R') {
      position = mod(position + distance, 100);
    }
    if (position == 0) {
      sum++;
    }
  }
  printf("%d\n", sum);
  fclose(fp);
  return 0;
}

File: src/2025/day1/day1.ts
// const text = await file.text()

const file = Bun.file("./2025_day1.txt");
const text = await file.text()

let position = 50;
let sum = 0;
function mod(a:number, b:number):number {
	let r = a % b;
	if (r < 0) {
		r += b;
	}
	return r;
}
for (const line of text.trim().split("\n")) {
	const direction = line[0]
	const distance = parseInt(line.slice(1), 10);
	if (direction == 'L') {
		position = mod(position - distance, 100)
	} else if (direction == 'R') {
		position = mod(position + distance, 100)
	}
	if (position == 0) {
		sum++;
	}
}
console.log(sum);